airbyte-cdk/python/airbyte_cdk/sources/streams/http/exceptions.py

```diff
@@ -9,7 +9,11 @@
 
 
 class BaseBackoffException(requests.exceptions.HTTPError):
-    pass
+    def __init__(self, request: requests.PreparedRequest, response: requests.Response):
+        self.request = request
+        self.response = response
+        message = f"\nRequest URL: {self.request.url}\nResponse Code: {self.response.status_code}\nResponse Text: {self.response.text}"
```

Comment:
```suggestion
        message = f"Request URL: {self.request.url}, Response Code: {self.response.status_code}, Response Text: {self.response.text}"
```
It could be unsafe to use newline separators, it could make the eventual downstream log parser fail to represent this as a single log message.